protographic/README.md

```diff
@@ -0,0 +1,162 @@
+  }
+);
+```
+
+### Using proto.lock.json for Deterministic Field Ordering
+
+Protographic supports deterministic field ordering for Protocol Buffer files through a lock file mechanism. This ensures backward compatibility across schema changes:
+
+```typescript
+import { compileGraphQLToProto } from '@wundergraph/protographic';
+
+// First generation with a new lock file
+const result1 = compileGraphQLToProto(initialSchema, {
+  serviceName: 'MyService',
+  lockFilePath: './proto.lock.json' // Creates lock file if it doesn't exist
+});
+
+// Later generation with schema changes but preserving field order
+const result2 = compileGraphQLToProto(updatedSchema, {
+  serviceName: 'MyService',
+  lockFilePath: './proto.lock.json' // Uses existing lock file
+});
+```
+
+When providing a `lockFilePath`, the function returns an object with both the proto definition and the lock data:
+
+```typescript
+const { proto, lockData } = compileGraphQLToProto(schema, { 
+  lockFilePath: './proto.lock.json' 
+});
+```
+
+If you prefer to manage the lock data directly without file I/O, you can use the lock data directly:
+
+```typescript
+import { compileGraphQLToProto, ProtoLock } from '@wundergraph/protographic';
+
+// First generation - creates initial lock data
+const result1 = compileGraphQLToProto(initialSchema, {
+  serviceName: 'MyService'
+});
+const proto1 = result1.proto;
+const lockData = result1.lockData;
+
+// Store the lock data however you want (database, state management, etc.)
+// ...
+
+// Later generation with the saved lock data
+const result2 = compileGraphQLToProto(updatedSchema, {
+  serviceName: 'MyService',
+  lockData: lockData // Use previously generated lock data
+});
+```
+
+The lock data records the order of:
+- Service methods
+- Message fields
+- Enum values
+- Implementers of interfaces
+- Members of unions
+
+New fields are always added at the end, maintaining backward compatibility with existing proto messages.
+
+### Generating Mapping Definitions
+
+```typescript
+import { compileGraphQLToMapping } from '@wundergraph/protographic';
+
+const graphqlSchema = `
+type Product @key(fields: "id") {
+  id: ID!
+  name: String!
+  price: Float!
+}
+
+type Query {
+  product(id: ID!): Product
+}
+`;
+
+const mappingOutput = compileGraphQLToMapping(
+  graphqlSchema, // String or GraphQLSchema object
+  'ProductService', // Service name
+);
+```
+
+## Conversion Rules
+
+Protographic follows a set of conventions when converting GraphQL SDL to Protocol Buffers. Please take a look at the following rules:
+
+1. GraphQL SDL to Mapping, see [SDL_MAPPING_RULES.md](SDL_MAPPING_RULES.md)
+2. GraphQL SDL to Proto, see [SDL_PROTO_RULES.md](SDL_PROTO_RULES.md)
+
+## Integration with graphql-go-tools
+
+Protographic integrates with [graphql-go-tools](https://github.com/wundergraph/graphql-go-tools) to enable seamless GraphQL-to-gRPC translation. This integration allows you to:
```

Comment:
probably results used by go-tools instead of lib itself?

Reply:
Done